client_test.go

```diff
@@ -0,0 +1,245 @@
+package httprequest_test
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/errgo.v1"
+
+	"github.com/juju/httprequest"
+	"github.com/julienschmidt/httprouter"
+)
+
+type clientSuite struct{}
+
+var _ = gc.Suite(&clientSuite{})
+
+var callTests = []struct {
+	about       string
+	client      httprequest.Client
+	req         interface{}
+	expectError string
+	expectCause interface{}
+	expectResp  interface{}
+}{{
+	about: "GET success",
+	req: &chM1Req{
```

Comment:
I started with clientHandlerM1Req, but the code was unreadable,
so I abbreviated clientHandler to ch.

Better suggestions welcome.